try print stacktrace for error

henrylhtsang opened the pull request:
Differential Revision: D69573525




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

Skylion007 commented on torch/_inductor/codegen/common.py:
```diff
@@ -2246,6 +2246,10 @@ def maybe_append_choice(
         except NotImplementedError as e:
+            log.info(
+                "Cannot Append Choice: %s",
+                stack_info=True if log.getEffectiveLevel() < logging.INFO else False,
```
Is this necessary? Stack_info=True should have no effect if the level is not at least log.info level? You could also pass exc_info for a more detailed trace.

henrylhtsang replied:
> Is this necessary? Stack_info=True should have no effect if the level is not at least log.info level? You could also pass exc_info for a more detailed trace.

I want to print the error with a higher logging level (info) and print also the stack trace at debug so to not spam people. Let me know what you think

Skylion007 replied:
Ah I see it's, suppose to be printer at debug only. Makes sense then.

henrylhtsang replied:
landing with `stack_info=log.getEffectiveLevel() < logging.INFO,` for now. Let me know if you prefer a bool cast

Skylion007 commented on torch/_inductor/codegen/common.py:
```diff
@@ -2246,6 +2246,10 @@ def maybe_append_choice(
             return None
         except NotImplementedError as e:
+            log.info(
+                "Cannot Append Choice: %s",
```
The choice % string isn't populated into this string, shouldn't it be?

henrylhtsang committed:
try print stacktrace for error (#147061)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/147061

Differential Revision: D69573525

henrylhtsang commented:
@pytorchbot merge -i
